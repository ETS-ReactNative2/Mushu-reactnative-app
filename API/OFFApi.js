import ProductService from "../Services/ProductService";

const lang = 'fr';
//TODO switch back to https
const apiUrl = 'https://' + lang + '.openfoodfacts.org';

export function getProductInfoFromApi(barcode) {
    const url = apiUrl + '/api/v0/product/' + barcode + '.json';
    return fetch(url)
        .then((response) => response.json())
        .catch((error) => console.error(error))
}

export function parseProductInfo(json, barcode) {
    if (json.status !== 0 && json.code && json.code.length > 0) {
        //TODO delete logs
        let jsonProduct = json.product;
        //     console.log("création du produit depuis le json");
        let product = ProductService.findProduct(jsonProduct, barcode);
        //   console.log("produit créé avec succès");
        ProductService.scan(product);

        return {
            _id: json.code,
            product_name: jsonProduct.product_name_fr,
            image_url: jsonProduct.image_url,
            quantity: jsonProduct.quantity,
            packaging: jsonProduct.packaging,
            brands: jsonProduct.brands,
            manufacturing_places: jsonProduct.manufacturing_places,
            categories: jsonProduct.categories,
            ingredients: jsonProduct.ingredients_text_with_allergens,
            allergens: jsonProduct.allergens_from_ingredients,
            nutrition_grades: jsonProduct.nutrition_grades,
            nova_group: jsonProduct.nova_group,
            allergens_ids: jsonProduct.allergens_tags

        };
    } else {
        return {};
    }
}

export function getAllergensFromApi() {
    const url = apiUrl + '/allergens.json';
    return fetch(url)
        .then((response) => response.json())
        .then((json) => {
            // console.log(json);
            let allergens = [];
            if (json.tags) {
                allergens = json.tags
                    .filter((obj => (obj.id !== obj.name) && obj.products > 50))
                    .map((obj) => {
                        return {
                            obj: {
                                id: obj.id,
                                name: obj.name},
                            name: obj.name,
                            _id: obj.id
                        }
                    });
            }
            return allergens;
        })
        .catch((error) => console.error(error))
}
