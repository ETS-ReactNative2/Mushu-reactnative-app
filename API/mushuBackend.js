const apiUrl = 'http://54.37.23.103/mushu';


export function getCFPFromBarcode(barcode) {
    const url = apiUrl + '/cfp?barcode=' + barcode;
    console.log("calling api: " + url);
    return fetch(url)
        .then((response) => {
            return response.json();
        })
        .catch() //network fail is handled in call in Product.js
}


export function formatProductJson(json) {
    return {
        barcode: json.barcode,
        name: json.name,
        imageUrl: json.image_url,
        CFPDensity: json.CFPDensity,
        totalCFP: json.TotalCFP,
        CFPUnit: json.CFPUnit,
        weight: json.weight,
        weightUnit: json.weightUnit,
        ingredients: json.ingredients
    };
}


export function getEquivFromCFP(cfp) {
    const url = apiUrl + '/equivalent?cfp=' + cfp;
    return fetch(url)
        .then((response) => {
            return response.json();
        })
        .catch() //network fail is handled in call in Product.js
}
