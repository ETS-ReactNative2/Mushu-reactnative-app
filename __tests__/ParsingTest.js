import React from 'react';

import renderer from 'react-test-renderer';
import sample from '../Helper/sample'
import {parseProductInfo} from '../API/OFFApi';


describe('Parsing', () => {
    jest.unmock('../API/OFFApi');
    it('should parse things right', () => {
        expect(parseProductInfo(sample)).toMatchSnapshot();
    });
})