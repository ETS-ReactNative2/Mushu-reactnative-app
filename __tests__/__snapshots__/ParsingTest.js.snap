// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parsing should parse things right 1`] = `Object {}`;
